src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.test.ts

```diff
@@ -0,0 +1,213 @@
+      const lastIncentiveBalance = BigNumber.from(1000);
+      const chainId = 1;
+      const flowType = "inflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("20");
+    });
+
+    it("should calculate the balancing fee for an outflow event", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(1000);
+
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should return a balanceFee of 0 if the amount is 0", () => {
+      const amount = BigNumber.from(0);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(1000);
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should return a balance fee 0 if lastIncentiveBalance is negative", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(-1000);
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should return a balance fee 0 if lastIncentiveBalance is 0", () => {
+      const amount = BigNumber.from(10);
+      const lastRunningBalance = BigNumber.from(1000);
+      const lastIncentiveBalance = BigNumber.from(0);
+      const chainId = 1;
+      const flowType = "outflow";
+      const fee = getEventFee(amount, flowType, lastRunningBalance, lastIncentiveBalance, chainId, defaultConfig);
+      expect(fee.balancingFee.toString()).toEqual("0");
+    });
+
+    it("should have an expected discount factor", () => {
+      const amount = toBNWei(10, 6);
+      const lastRunningBalance = toBNWei(2000, 6);
+      const lastIncentiveBalance = toBNWei(100, 6);
+      const chainId = 1;
+      const flowType = "outflow";
+      const config = new MockUBAConfig();
+
+      config.setBalancingFeeCurve(chainId.toString(), [
+        [toBNWei(0, 6), toBNWei(0, 0)],
+        [toBNWei(1, 6), toBNWei(0.2, 18)],
+      ]);
+
+      const discountFactorBalancingFee = getEventFee(
+        amount,
+        flowType,
+        lastRunningBalance,
+        lastIncentiveBalance,
+        chainId,
+        config
+      ).balancingFee;
+
+      const nonDiscountedFee = getEventFee(
+        amount,
+        flowType,
+        lastRunningBalance,
```

Comment:
```suggestion
        // This fee is "non-discounted" because the `lastIncentiveBalance` is increased such that the balancing fee no longer needs to be discounted
        lastRunningBalance,
```

I think we should not set `lastIncentiveBalance = lastRunningBalance` but should set it explicitly to something like `toBNWei("1000")` for example so that its very clear how this test is working. Also comments should be added here.